### PR TITLE
Newweb

### PR DESCRIPTION
**Pull request by olpapina**

### CODE REVIEW

**Review by TatsianaLohinava — COMMENTED** [review]
Code looks fine to me

File: src/main/java/com/solvd/webtesting/elements/ProductMenuBar.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                .findFirst()
+                .orElseThrow();
+        requiredProduct.click();
+        return new ProductPage(getDriver());
+    }
+
+    public List<ExtendedWebElement> getItemProducts() {
+        return itemProducts;
+    }
+}
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/elements/ProductMenuBar.java]: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/elements/SearchSection.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public ExtendedWebElement getSearchButton() {
+        return searchButton;
+    }
+
+    public void setSearchButton(ExtendedWebElement searchButton) {
+        this.searchButton = searchButton;
+    }
+}
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/elements/SearchSection.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/elements/SideBar.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return minPriceField;
+    }
+
+    public ExtendedWebElement getMaxPriceField() {
+        return maxPriceField;
+    }
+
+    public List<ExtendedWebElement> getCheckboxes() {
+        return checkboxes;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/elements/SideBar.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/BrandProductPage.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        this.brandProducts = brandProducts;
+    }
+
+    public List<ExtendedWebElement> getProductPrices() {
+        return productPrices;
+    }
+
+    public void setProductPrices(List<ExtendedWebElement> productPrices) {
+        this.productPrices = productPrices;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/BrandProductPage.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        return accountMenuAfterLogin;
+    }
+
+    public ExtendedWebElement getUserToolsTitle() {
+        return userToolsTitle;
+    }
+
+    public ProductMenuBar getProductMenuBar() {
+        return ProductMenuBar;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/HomePage.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/LoginPage.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    public HomePage clickEnterButton() {
+        if (enterButton.isElementPresent(5L) & enterButton.isClickable()) {
+            enterButton.click();
+        }
+        return new HomePage(getDriver());
+    }
+
+    public ExtendedWebElement getRegistrationButton() {
+        return registrationButton;
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/LoginPage.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/RegistrationPage.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        return new RegistrationPrivacyPage(getDriver());
+    }
+
+    public ExtendedWebElement getEmailField() {
+        return emailField;
+    }
+
+    public ExtendedWebElement getContinueButton() {
+        return continueButton;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/RegistrationPage.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/RegistrationPrivacyPage.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    public SuccessRegistrationPage clickPrivacyButton() {
+        if (privacyButton.isElementPresent(5L) & privacyButton.isClickable()) {
+            privacyButton.click();
+        }
+        return new SuccessRegistrationPage(getDriver());
+    }
+
+    public ExtendedWebElement getPrivacyButton() {
+        return privacyButton;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/RegistrationPrivacyPage.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/ResultPage.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+    public List<String> getFullNames() {
+        return searchResultItems.stream()
+                .map(ExtendedWebElement::getText)
+                .collect(Collectors.toList());
+    }
+
+    public List<ExtendedWebElement> getSearchResultItems() {
+        return searchResultItems;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/ResultPage.java]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/page/SuccessRegistrationPage.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    @FindBy(css = ".styles_successTitle__1IDi2")
+    private ExtendedWebElement successRegistration;
+
+    public SuccessRegistrationPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public ExtendedWebElement getSuccessRegistration() {
+        return successRegistration;
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/SuccessRegistrationPage.java:18]: The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.

File: src/main/java/com/solvd/webtesting/utils/CapabilityFactory.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+            case "firefox":
+                mutableCapabilities = new FirefoxCapabilities().getCapability("FirefoxTest");
+                break;
+            case "chrome":
+                mutableCapabilities = new ChromeCapabilities().getCapability("ChromeTest");
+                break;
+            default:
+                LOGGER.info("browser : " + browser + " is invalid, Launching Chrome as browser of choice..");
+                mutableCapabilities = new ChromeCapabilities().getCapability("ChromeTest");
+                break;
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/utils/CapabilityFactory.java]: Use LOGGER.error

File: src/main/java/com/solvd/webtesting/page/HomePage.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    @FindBy(xpath =  "//*[contains(@class,'styles_headerReactWrapper__3U15a') and contains(@class,'styles_headerReactWrapperAlt__3bM5y')]")
+    private com.solvd.webtesting.elements.ProductMenuBar ProductMenuBar;
+
+    @FindBy(xpath = "//*[contains(@class,'styles_headerReactLine__IWu_8') and contains(@class,'styles_headerReactLineSecond__tp1U0')]")
+    private com.solvd.webtesting.elements.SearchSection SearchSection;
+
+    public HomePage(WebDriver driver) {
+        super(driver);
+        setPageURL("");
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/webtesting/page/HomePage.java]: Why did you set empty URL?

Comment (olpapina) [reply src/main/java/com/solvd/webtesting/page/HomePage.java]: I saw in carina demo URL set as : 
        setPageAbsoluteURL(R.CONFIG.get(Configuration.Parameter.URL.getKey()));
      But during Laba we used to environment configurations: 
      env=PROD
      PROD.base=https://21vek.by/
       and if I understand correctly that carina pulls up the base url or am I mistaken?

File: src/main/java/com/solvd/webtesting/utils/CapabilityFactory.java (hunk 11 above)
Comment (olpapina) [reply src/main/java/com/solvd/webtesting/utils/CapabilityFactory.java]: resolved

File: src/main/java/com/solvd/webtesting/page/SuccessRegistrationPage.java (hunk 10 above)
Comment (olpapina) [reply src/main/java/com/solvd/webtesting/page/SuccessRegistrationPage.java:18]: resolved for all unused getElements

**Review by marinagavrilenko — APPROVED** [review]
LGTM. You merge this PR
Good work!

**Pull request merged**